fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java

```diff
@@ -1758,22 +1769,27 @@ public int getAsInt() {
                     }
                 } else if (alterClause instanceof ModifyColumnClause) {
                     // modify column
-                    processModifyColumn((ModifyColumnClause) alterClause, olapTable, indexSchemaMap);
+                    boolean clauseCanLigthSchemaChange = processModifyColumn((ModifyColumnClause) alterClause, olapTable, indexSchemaMap);
                     lightSchemaChange = false;
+                    if (clauseCanLigthSchemaChange == false) {
+                        lightSchemaChangeWithVarcharLength = false;
+                    }
                 } else if (alterClause instanceof ReorderColumnsClause) {
                     // reorder column
                     processReorderColumn((ReorderColumnsClause) alterClause, olapTable, indexSchemaMap);
                     lightSchemaChange = false;
+                    lightSchemaChangeWithVarcharLength = false;
```

Comment:
can we reuse lightSchemaChange instead of add a new flag lightSchemaChangeWithVarcharLength?